Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/RidePayment.hs

```diff
@@ -161,35 +161,15 @@ postPaymentAddTip ::
     API.Types.UI.RidePayment.AddTipRequest ->
     Environment.Flow APISuccess
   )
-postPaymentAddTip (mbPersonId, merchantId) rideId tipRequest = do
+postPaymentAddTip (_mbPersonId, merchantId) rideId tipRequest = do
   Redis.whenWithLockRedis addTipLockKey 60 $ do
-    personId <- mbPersonId & fromMaybeM (PersonNotFound "No person found")
-    person <- runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
     ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideNotFound rideId.getId)
     unless (ride.status == Domain.Types.Ride.COMPLETED) $
       throwError $ RideInvalidStatus ("Ride is not completed yet." <> Text.pack (show ride.status))
     fareBreakups <- runInReplica $ QFareBreakup.findAllByEntityIdAndEntityType rideId.getId Domain.Types.FareBreakup.RIDE
     when (any (\fb -> fb.description == tipFareBreakupTitle) fareBreakups) $ throwError $ InvalidRequest "Tip already added"
-    customerPaymentId <- person.customerPaymentId & fromMaybeM (PersonFieldNotPresent "customerPaymentId")
```

Comment:
if the tip is added after the scheduler run, then tip should be charged separately